site/app/models/Email.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace app\models;
+use app\libraries\Core;
+use app\libraries\Utils;
+
+/**
+ * Class Email
+ *
+ * @method void     setSubject($sub)
+ * @method void     setBody($bod)
+ * @method void     setRecipient($recip)
+
+ * @method string   getSubject()
+ * @method string   getBody()
+ * @method string   getRecipient()
+ */
+class Email extends AbstractModel {
+    /** @property @var string Subject line of email */
+    protected $subject;
+    /** @property @var string Body of email */
+    protected $body;
+    /** @property @var string Intended receiver of email */
+    protected $recipient;
+
+  /**
+   * Email constructor.
+   *
+   * @param Core  $core
+   * @param array $details
+   */
+
+    public function __construct(Core $core, $details=array()) {
+        parent::__construct($core);
+        if (count($details) == 0) {
+            return;
+        }
+
+        $this->setRecipient($details["recipient"]);
+        switch($details["type"]){
+            case 'forum_announcement':
+                $this->handleForumAnnouncement($details["subject"], $details["body"]);
+                break;
+            case 'seating_assignment':
+                $this->handleSeatingAssignment($details["subject"], $details["body"]);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private function handleForumAnnouncement($subject, $body) {
+        $formatted_subject = $this->formatSubject($subject);
+        $formatted_body = "An Instructor/TA made an announcement in the Submitty discussion forum:\n\n".$this->formatBody($body);
+
+        $this->setSubject($formatted_subject);
+        $this->setBody($formatted_body);
+    }
+
+    private function handleSeatingAssignment($subject, $body) {
+        $formatted_body = $this->formatBody($body);
```

Comment:
Why does this function not use formatSubject? Aren't these emails related to a specific course?

Reply:
So in the case of seating assignments we preview the subject/email to the sender, so I have the [Submitty $course_name] already injected. However, this can be changed/refactored for consistency reasons.

Reply:
Yeah, we should refactor this so that we don't have two different methods where that bit gets injected into the title, I don't really have much of a preference on whether that's before we construct the model, or if it's part of a factory method within the model (e.g. `createCourseEmail`). Thinking about this, we can probably move the bit about "Instructor/TA has made an announcement..." into where the the email model is constructed for announcements. I don't think we will re-use that bit beyond for just announcements, whereas as `createCourseEmail` factory method would take in a $core, title, body and then prepend "[Submitty {$course_name}]" to the title, and then `return new Email($core, $title, $body);`.